rust/gui-client/src-tauri/src/client/gui.rs

```diff
@@ -489,11 +492,15 @@ impl Controller {
     async fn handle_request(&mut self, req: ControllerRequest) -> Result<(), Error> {
         match req {
             Req::ApplySettings(settings) => {
+                let filter =
+                    tracing_subscriber::EnvFilter::try_new(&self.advanced_settings.log_filter)
+                        .context("Couldn't parse new log filter directives")?;
                 self.advanced_settings = settings;
-                // TODO: Update the logger here if we can. I can't remember if there
-                // was a reason why the reloading didn't work.
-                tracing::info!(
-                    "Applied new settings. Log level will take effect at next app start."
+                self.log_filter_reloader
+                    .reload(filter)
+                    .context("Couldn't reload log filter")?;
+                tracing::debug!(
+                    "Applied new settings. Log level will take effect immediately for the GUI and later for the IPC service."
```

Comment:
Can we send an IPC message and reload the filter there too?

Reply:
Yeah that's the plan. Right now the IPC connection is only up when we're signed in, so I'm going to change that so the GUI connects immediately when it starts.

Reply:
That will also unblock #5453

Reply:
Sounds great :)